docs/reference/TensorFlow.md

```diff
@@ -1,405 +1,394 @@
 
-3.  Use new trained models to recognize different words
-
-# Download and build the sample application
-
-**Install Arm toolchain and Mbed CLI**
--   Download [Arm cross compilation](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads) toolchain. Select the correct toolchain for the OS that your computer is running. For Windows users, if you have already set up the Linux virtual environment, install the toolchain there.
-
--   To build and deploy the application, we will use the [Mbed CLI](https://github.com/ARMmbed/mbed-cli). We recommend that you install Mbed CLI with our installer. If you need more customization, you can perform a manual install. Although this is not recommended.  
-
-    If you do not already have Mbed CLI installed, download the installer:  
-
-    [Mac installer](https://github.com/ARMmbed/mbed-cli-osx-installer/releases/tag/v0.0.10)
--   After Mbed CLI is installed, tell Mbed where to find the Arm embedded toolchain.  
-  ```    
- mbed config -G GCC_ARM_PATH <path_to_your_arm_toolchain>/bin
- ```
-
-**Important:** We recommend running the following commands from inside the Mbed CLI terminal that gets launched with the Mbed CLI Application. This is because it will be much quicker to set up, because it resolves all your environment dependencies automatically.
+1. Navigate into the `mbed` folder:
 
+    ```
+    cd tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed
+    ```
 
-# Build and compile micro speech example
+1. Execute the following (on an environment that runs Python 2.7):
 
-Navigate to the directory where you keep code projects. Run the following command to download TensorFlow Lite source code.
+    ```
+    mbed config root .
 
-```git clone https://github.com/tensorflow/tensorflow.git```
+    mbed deploy
 
-While you wait for the project to download, let’s explore the project files on [GitHub](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/micro/examples/micro_speech) and learn how this TensorFlow Lite for Microcontrollers example works.
+    mbed compile -m K66F -t GCC_ARM
+    ```
 
-The code samples audio from the microphone on the K66F. The audio is run through a Fast Fourier transform to create a spectrogram. The spectrogram is then fed into a pre-trained machine learning model. The model uses a [convolutional neural network](https://developers.google.com/machine-learning/practica/image-classification/convolutional-neural-networks) to identify whether the sample represents either the command “yes” or “no”, silence, or an unknown input. We will explore how this works in more detail later in the guide.
+    <!--The install page for Mbed CLI now requires 3.7. Why are we asking for 2.7?-->
+    <!--need to explain what `config root .` and `deploy` do and why we need them - neither one is part of a standard workflow where you use Mbed CLI to import an application, so this is a special case-->
```

Comment:
From what I understand, `mbed config root .` sets the current mbed-os folder as the root (meaning `mbed` command actually knows what Mbed OS project we want to run the following commands on) which then allows us to run `mbed deploy` which can pull the libraries needed (in our case in tensorflow/tensorflow/lite/micro/tools/make/gen/mbed_cortex-m4/prj/micro_speech/mbed/mbed-os/requirements.txt) and installs them. I was not aware that this is not a standard workflow, this is the workflow presented in the [original guide hosted on developer.arm.com](https://developer.arm.com/solutions/machine-learning-on-arm/developer-material/how-to-guides/build-arm-cortex-m-voice-assistant-with-google-tensorflow-lite/download-and-build-the-sample-application). If you know of a better way of doing this I'd be glad to explore that workflow and update the guide.

Reply:
@donatieng can you confirm/deny?

Reply:
See above, I think this is a shortcoming of the Makefile here

Reply:
are we good here?